src/Offer/AbstractGeoCoordinatesCommandHandler.php

```diff
@@ -46,33 +51,52 @@ public function __construct(
         $this->defaultAddressFormatter = $defaultAddressFormatter;
         $this->fallbackAddressFormatter = $fallbackAddressFormatter;
         $this->geocodingService = $geocodingService;
+        $this->logger = new NullLogger();
     }
 
     /**
      * @param AbstractUpdateGeoCoordinatesFromAddress $updateGeoCoordinates
      */
     protected function updateGeoCoordinatesFromAddress(AbstractUpdateGeoCoordinatesFromAddress $updateGeoCoordinates)
     {
-        $coordinates = $this->geocodingService->getCoordinates(
-            $this->defaultAddressFormatter->format(
-                $updateGeoCoordinates->getAddress()
-            )
+        $offerId = $updateGeoCoordinates->getItemId();
+
+        $exactAddress = $this->defaultAddressFormatter->format(
+            $updateGeoCoordinates->getAddress()
         );
 
+        $coordinates = $this->geocodingService->getCoordinates($exactAddress);
+
         if ($coordinates === null) {
-            $coordinates = $this->geocodingService->getCoordinates(
-                $this->fallbackAddressFormatter->format(
-                    $updateGeoCoordinates->getAddress()
+            $fallbackAddress = $this->fallbackAddressFormatter->format(
+                $updateGeoCoordinates->getAddress()
+            );
+
+            $this->logger->debug(
+                sprintf(
+                    "Could not find coordinates for exact address '%s', trying '%s' instead for offer id %s.",
+                    $exactAddress,
+                    $fallbackAddress,
+                    $offerId
                 )
             );
+
+            $coordinates = $this->geocodingService->getCoordinates($fallbackAddress);
+
+            if (!is_null($coordinates)) {
+                $this->logger->debug(
+                    "Found coordinates for fallback address '$fallbackAddress' for offer id $offerId"
+                );
+            }
```

Comment:
We could have made this an `else` for the `if ($coordinates === null)` below, but we didn't because we only want to log this when we find coordinates for the fallback address. Not when we find coordinates for the full address, because we're not interested in those cases.